### PR TITLE
Handle codegen-ed versioned features [CHA-3409]

### PR DESCRIPTION
**Pull request by rooftoofwoof**
This PR adds support for codegen-ed versioned features

### CODE REVIEW

File: features.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 
 type Feature struct {
 	Fqn string
+
+	isWindowedPseudoFeature *bool
```
Comment (rooftoofwoof) [inline features.go:13]: remove

File: init_features.go [hunk 2]
```diff
@@ -12,10 +13,14 @@ func InitFeatures[T any](t *T) error {
 	return initFeatures(structValue, "", make(map[string]bool), nil)
 }
 
+type initContext struct {
```
Comment (rooftoofwoof) [inline init_features.go:16]: remove

File: init_features.go [hunk 3]
```diff
@@ -12,10 +13,14 @@ func InitFeatures[T any](t *T) error {
...
+	IsParentWindowedFeature bool
+}
+
 // initFeatures is a recursive function that initializes all features
 // in the struct that is passed in. Each feature is initialized as
 // a pointer to a Feature struct with the appropriate FQN.
-func initFeatures(structValue reflect.Value, fqn string, visited map[string]bool, fieldMap fqnToField) error {
+func initFeatures(structValue reflect.Value, fqn string, visited map[string]bool, fieldMap fqnToFields) error {
```
Comment (rooftoofwoof) [inline init_features.go:23]: fieldsMap

File: init_features_test.go [hunk 4]
```diff
@@ -28,6 +33,17 @@ type goUser struct {
...
+	// Versioned features
+	Grade   *int `versioned:"default(2)"`
+	GradeV1 *int `versioned:"true"`
+	GradeV2 *int `versioned:"true"`
+
+	// Windowed features
+	AvgSpend map[string]*float64 `windows:"1m,5m,1h"`
+
+	// Dataclass features
+	LatLng *goLatLng
```
Comment (rooftoofwoof) [inline init_features_test.go:46]: Add the dataclass struct tag

**Review by emarx — APPROVED** [review]

**Pull request merged**